x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/rollup/v2/RollupShardIndexer.java

```diff
@@ -268,7 +268,6 @@ public void collect(int docId, long owningBucketOrd) throws IOException {
                             + "] -> ["
                             + timestampFormat.format(timestamp)
                             + "]";
-                    lastTsid = BytesRef.deepCopyOf(tsid);
```

Comment:
This tsid copy had been originally added to support the assertions and debug the code. I would rather remove the assertions altogether than keep the copy at every iteration. 

Thanks for fixing this